watcher/kubernetes/pods.go

```diff
@@ -16,38 +20,59 @@ import (
 
 // PodsManager defined pods manager struct
 type PodsManager struct {
-	client        kubernetes.Interface
-	eventManager  *EventsManager
-	pvcManager    *PvcManager
-	Watch         chan WatchData
-	podsFirstInit map[string]bool
-	mutex         *sync.RWMutex
+	client                 kubernetes.Interface
+	eventManager           *EventsManager
+	pvcManager             *PvcManager
+	Watch                  chan WatchData
+	podsFirstInit          map[string]bool
+	podsLogsFirstInit      map[string]bool
+	mutexPodsFirstInit     *sync.RWMutex
+	mutexPodsLogsFirstInit *sync.RWMutex
+	absoluteLogsPodPath    string
 }
 
 // NewPodsManager create new pods instance
-func NewPodsManager(kubernetesClientset kubernetes.Interface, eventManager *EventsManager, pvcManager *PvcManager) *PodsManager {
+func NewPodsManager(kubernetesClientset kubernetes.Interface, eventManager *EventsManager, pvcManager *PvcManager, absoluteLogsPodPath string) *PodsManager {
 	return &PodsManager{
-		client:        kubernetesClientset,
-		eventManager:  eventManager,
-		pvcManager:    pvcManager,
-		podsFirstInit: map[string]bool{},
-		mutex:         &sync.RWMutex{},
-		Watch:         make(chan WatchData),
+		client:                 kubernetesClientset,
+		eventManager:           eventManager,
+		pvcManager:             pvcManager,
+		podsFirstInit:          map[string]bool{},
+		podsLogsFirstInit:      map[string]bool{},
+		mutexPodsFirstInit:     &sync.RWMutex{},
+		mutexPodsLogsFirstInit: &sync.RWMutex{},
+		Watch:                  make(chan WatchData),
+		absoluteLogsPodPath:    absoluteLogsPodPath,
 	}
 }
 
 // storePodFirstInit will set if some pod appears for the first time true == first time
 func (pm *PodsManager) storePodFirstInit(key string, val bool) {
```

Comment:
I'm not sure all of these locks are necessary.. at least at this level.. let's discuss it online when you have time